src/target/riscv/riscv.c

```diff
@@ -1011,25 +1011,29 @@ static int maybe_add_trigger_t2_t6_for_wp(struct target *target,
 		struct trigger *trigger, struct match_triggers_tdata1_fields fields)
 {
 	RISCV_INFO(r);
-	int ret = ERROR_FAIL;
+	int ret = ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
 
 	if (trigger->length > 0) {
 		/* Setting a load/store trigger ("watchpoint") on a range of addresses */
-
-		if (r->enable_napot_trigger && can_use_napot_match(trigger)) {
-			LOG_TARGET_DEBUG(target, "trying to setup NAPOT match trigger");
-			struct trigger_request_info napot = {
-				.tdata1 = fields.common | fields.size.any |
-					fields.chain.disable | fields.match.napot,
-				.tdata2 = trigger->address | ((trigger->length - 1) >> 1),
-				.tdata1_ignore_mask = fields.tdata1_ignore_mask
-			};
-			ret = try_setup_single_match_trigger(target, trigger, napot);
-			if (ret != ERROR_TARGET_RESOURCE_NOT_AVAILABLE)
-				return ret;
+		if (can_use_napot_match(trigger)) {
+			if (r->wp_enable_napot_trigger) {
+				LOG_TARGET_DEBUG(target, "trying to setup NAPOT match trigger");
+				struct trigger_request_info napot = {
+					.tdata1 = fields.common | fields.size.any |
+						fields.chain.disable | fields.match.napot,
+					.tdata2 = trigger->address | ((trigger->length - 1) >> 1),
+					.tdata1_ignore_mask = fields.tdata1_ignore_mask
+				};
+				ret = try_setup_single_match_trigger(target, trigger, napot);
+				if (ret != ERROR_TARGET_RESOURCE_NOT_AVAILABLE)
+					return ret;
+			} else {
+				LOG_TARGET_DEBUG(target, "NAPOT match triggers are disabled for watchpoints. "
+							 "Use set_wp_enable_napot_trigger to enable it.");
```

Comment:
Please, update the name of the command also in the debug print:

```suggestion
				LOG_TARGET_DEBUG(target, "NAPOT match triggers are disabled for watchpoints. "
							 "Use set_enable_trigger_type to enable it.");
```

Reply:
Addressed.